pkg/kv/kv.go

```diff
@@ -766,3 +770,87 @@ const (
 	// RCCheckTS stands for 'read consistency read with ts check'.
 	RCCheckTS
 )
+
+// ResourceGroupTagBuilder is used to build the resource group tag for a kv request.
+type ResourceGroupTagBuilder struct {
+	sqlDigest  *parser.Digest
+	planDigest *parser.Digest
+	accessKey  []byte
+}
+
+// NewResourceGroupTagBuilder creates a new ResourceGroupTagBuilder.
+func NewResourceGroupTagBuilder() *ResourceGroupTagBuilder {
+	return &ResourceGroupTagBuilder{}
+}
+
+// SetSQLDigest sets the sql digest for the request.
+func (b *ResourceGroupTagBuilder) SetSQLDigest(digest *parser.Digest) *ResourceGroupTagBuilder {
```

Comment:
It seems not common to return self for SetXXX function in tidb's existing go code

Reply:
Here example are: https://github.com/pingcap/tidb/blob/master/pkg/distsql/request_builder.go#L177-L200

Reply:
OK